### PR TITLE
feat(mojaloop/#2092): upgrade nodeJS version for core services

### PR DESCRIPTION
**Pull request by mdebarros**
- updated dependencies
- upgraded nodejs to latest lts for https://github.com/mojaloop/project/issues/2745
  - added .nvmrc
  - updated circleci config
- updated ci-cd for automated release
- updated ci-cd ubuntu image to the latest available
- fixed auto-commit messages on CI-CD release to correctly reference the mojaloop/issue repo by adding a standard-release override to the package.json
- re-structured project code to align to core Mojaloop repositories to enable sharing of the circleci configuration to run standard CI-CD flows
- fixed unit test paths to reflect the project re-structuring
- added `test-lint` to CI-CD flows
- added `test-dependencies` to CI-CD flows
- added slack messages to GH, NPM publish jobs with appropriate error messages. We can now track GH and NPM jobs independently via slack as a result. Previously only GH releases would be announced, with the NPM release potentially being "swallowed" until someone noticed.
- added `snapshot` script to package.json. This will auto-bump the npm version with `snapshot.#` and commit both the package/package-lock files. Super useful for working with snapshot releases. This will also create an associated snapshot tag, and all the developer would need to do is push that tag for the snapshot build to be released. This can be done from any branch.
- updated CircleCI Orb for Slack notifications to the latest version using standard templates: https://github.com/mojaloop/ci-config/tree/master/slack-templates

Notes: 
- npm-audit-resolver v3.0.0-7 is a candidate release to resolve compatibility with npm v7+ as described in https://github.com/naugtur/npm-audit-resolver/issues/34. This will need to be addressed going forward as `npm run audit:resolve` (i.e. `resolve-audit`) is currently not functioning. As a work-around, we need to manually run the following command `npm audit fix`. The `npm run audit:check` (i.e. `check-audit`) still works as expected.

BREAKING CHANGE: major version bump for node v16 LTS support, and re-structuring of project directories to align to core Mojaloop repositories!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,181 +36,175 @@ defaults_license_scanner: &defaults_license_scanner
...
-      -d "{
-      \"text\": \"*${CIRCLE_PROJECT_REPONAME}* - Release \`${CIRCLE_TAG}\`: https://github.com/mojaloop/${CIRCLE_PROJECT_REPONAME}/releases/tag/${CIRCLE_TAG}\"
-    }"
+    git config user.email ${GIT_CI_EMAIL}
+    git config user.name ${GIT_CI_USER}
+
+##
+# Executors
+#
+# CircleCI Executors
```
Comment (lewisdaly) [inline .circleci/config.yml:72]: These lines can be tidied up

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,181 +36,175 @@ defaults_license_scanner: &defaults_license_scanner
...
+    executor: default-docker
     steps:
       - run:
           name: Install general dependencies
           command: *defaults_Dependencies
       - checkout
       - run:
           name: Update NPM install
-          command: cd src && npm ci
+          command: npm install
```
Comment (lewisdaly) [inline .circleci/config.yml:95]: Shouldn't we use `npm ci`? `npm install` doesn't respect the `package-lock.json`

**Review by lewisdaly — CHANGES_REQUESTED** [review]

**Comment by elnyry-sam-k** [comment]
Looking good Miguel, just one minor change from me.. Either mark the PR as breaking change "!" or upgrade package version to v17 or as relevant

**Comment by mdebarros** [comment]
> Looking good Miguel, just one minor change from me.. Either mark the PR as breaking change "!" or upgrade the package version to v17 or as relevant

Hey @elnyry-sam-k is already marked as breaking by adding this into the commit message:

```
BREAKING CHANGE: ....
```

File: .circleci/config.yml (hunk 2 above)
Comment (mdebarros) [reply .circleci/config.yml:95]: Good point

File: .circleci/config.yml (hunk 1 above)
Comment (lewisdaly) [reply .circleci/config.yml:72]: My mistake - the github diff made this look funny.

**Review by lewisdaly — APPROVED** [review]

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**